File: 3783.mirror-distance-of-an-integer.py
#
# @lc app=leetcode id=3783 lang=python
#
# [3783] Mirror Distance of an Integer
#

# @lc code=start
class Solution(object):
    def mirrorDistance(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n < 10:
            return 0
        else:
            original = n
            rev = 0
            while n > 0:
                digit = n % 10
                rev *= 10
                rev += digit
                n //= 10
            return abs(original - rev)
# @lc code=end

File: 3783.mirror-distance-of-an-integer.java
/*
 * @lc app=leetcode id=3783 lang=java
 *
 * [3783] Mirror Distance of an Integer
 */

// @lc code=start
class Solution {
    public int mirrorDistance(int n) {
        int rev = 0;
        int original = n;
        if (n < 10) {
            return 0;   
        }
        else {
        while (n > 0) {
            int digit = n % 10;
            rev *= 10;
            rev += digit;
            n = n /10;
        }
        }
        return Math.abs(rev - original);
    }
}
// @lc code=end

